### ISSUE TITLE
Can terratests be run against existing AWS resources created with terraform?

### ISSUE THREAD

**Issue by artisood** [gruntwork-io/terratest#519]
Are there any examples to accomplish testing  with terratest the AWS infrastructure that was created before and now is being updated? How can terratest know about the details of the existing infrastructure?

**Comment by brikis98** [comment]
I don't think we have examples of it, but it would be more or less identical to the examples out there, such as the [hello world example](https://github.com/gruntwork-io/terratest/blob/master/test/terraform_hello_world_example_test.go), except perhaps instead of calling `terraform.InitAndApply`, you'd only call `terraform.Init`, as the infra is already  deployed.

**Comment by brikis98** [comment]
> I'm new to terratest, but it seems to pick up the existing state file. And then go ahead and delete everything in it, and recreate, and then delete again.

Terratest runs the Terraform code you tell it to run! So if you pointed it at a Terraform module that had a backend configured with a state file, then it'll run against that state. If you want to override that, see the backend config options in `terraform.Options`.

**Comment by tonyszhang** [comment]
Yeah, you would need to create and store multiple tfstate files if you want to test entire infra without affecting existing stack. You would also need to namespace the actual AWS resources to avoid collision.

**Comment by james03160927** [comment]
inactive for the past 3 years. Closing this issue as there's no more interest. Feel free to reopen this if we still need this.
